### PR TITLE
Minor improvements in isotovideo - Alternative to #2032

### PR DESCRIPTION
**Pull request by okurz**
* Use log functions directly in isotovideo
* Simplify isotovideo with Mojo::File
* Use more signatures in isotovideo
* Move import statements to top in isotovideo

### CODE REVIEW

**Review by Martchus — APPROVED** [review]

**Comment by perlpunk** [comment]
Failure: https://github.com/os-autoinst/os-autoinst/runs/6153111082?check_suite_focus=true#step:3:1061
```
7: # '
1062
7: #     doesn't match '(?^u:save_tmp_file returned expected file)'
1063
7: 
1064
7: #   Failed test 'Result in testresults/result-modify_and_upload_file.json is ok'
1065
7: #   at ./99-full-stack.t line 69.
1066
7: #          got: 'fail'
1067
7: #     expected: 'ok'
1068
7: # Looks like you failed 2 tests of 26.
1069
7: [07:19:01] ./99-full-stack.t ........................ 
1070
7: Dubious, test returned 2 (wstat 512, 0x200)
1071
7: Failed 2/26 subtests 
1072
7: [07:19:01]
1073
7: 
1074
7: Test Summary Report
1075
7: -------------------
1076
7: ./99-full-stack.t                      (Wstat: 512 Tests: 26 Failed: 2)
1077
7:   Failed tests:  12, 17
```
I cannot reproduce this locally.

**Comment by okurz** [comment]
Yes, I have seen the same in #2032 sporadically as well but could not yet reproduce this problem locally.

File: isotovideo [hunk 1]
```diff
@@ -298,7 +293,7 @@ $command_handler = OpenQA::Isotovideo::CommandHandler->new(
     cmd_srv_fd => $cmd_srv_fd,
     backend_fd => $backend_process->channel_in,
 );
-$command_handler->on(tests_done => sub {
+$command_handler->on(tests_done => sub (@) {
```
Comment (kalikiana) [inline isotovideo:296]:
```suggestion
$command_handler->on(tests_done => sub ($e, @args) {
```

c.f. https://metacpan.org/pod/Mojo::EventEmitter#on

**Comment by okurz** [comment]
regarding the failure in t/99-full-stack.t, I can't reproduce it with `count_fail_ratio prove -I. t/99-full-stack.t`, that's 20/20 passed runs. Can anyone else try to reproduce?

File: isotovideo (hunk 1 above)
Comment (Martchus) [reply isotovideo:296]: I would name it as this is the officially documented signature. The additional information doesn't hurt. There are no "unused variable" warnings in Perl. In other languages where you would get a warning it is even practice to use things like [Q_UNUSED](https://doc.qt.io/qt-5/qtglobal.html#Q_UNUSED).

File: isotovideo [hunk 2]
```diff
@@ -256,10 +256,7 @@ my $testfd;
 ($testprocess, $testfd) = autotest::start_process();
 
 init_backend();
-
-open(my $fd, ">", "os-autoinst.pid");
-print $fd "$$\n";
-close $fd;
+path('os-autoinst.pid')->spurt("$$");
```
Comment (kalikiana) [inline isotovideo:259]: Note that this doesn't add a newline. Not sure if we rely on that

**Comment by Martchus** [comment]
I'm currently on run 19 locally and haven't encountered the problem so far. However, apparently the very same failure is happening also on https://github.com/os-autoinst/os-autoinst/pull/2041 (see https://github.com/os-autoinst/os-autoinst/runs/6191599180) which only changes the spec file and documentation. So it is definitely not related to either of our PRs.

**Comment by Martchus** [comment]
I've seen the issue also on more PRs so I suppose it is a global issue. I've been creating https://github.com/os-autoinst/os-autoinst/pull/2042 to test my libslirp0 theory.

**Comment by perlpunk** [comment]
Yeah, i've also verified this on github actions on master

**Comment by kalikiana** [comment]
Let's wait for #2042 to be merged and then `@mergifyio rebase` this one

**Comment by Martchus** [comment]
@mergifyio rebase

**Pull request merged**